src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs

```diff
@@ -204,6 +204,15 @@ private static EnumConverter CreateEnumConverter(Type type)
 
         private static Hashtable ExtendedPropertyCache => LazyInitializer.EnsureInitialized(ref s_extendedPropertyCache, () => new Hashtable());
 
+        /// <summary>Clear the global caches this maintains on top of reflection.</summary>
+        internal static void ClearReflectionCaches()
+        {
+            Volatile.Write(ref s_propertyCache, null);
```

Comment:
The Volatile here means that other threads will see these nulled in the order written here, right? Is that the purpose of it, and why is that important? Just curious.

Reply:
The volatiles here aren't really necessary.  I can remove them.  I'd added them just because all access to them in existing usage is via volatile, so I figured I'd be consistent.